### DISCUSSION TITLE
why does HTML only output mode even exist?

### DISCUSSION THREAD

**Discussion by SamKacer** [KaTeX/KaTeX#2576] [General] [answered]
I just came across this repo and it is a super cool piece of tech.

One thing that really puzzles me and kind of irritates me is why there is an HTML only output mode? Selecting that mode the user is basically saying " I just want the visual rendering of the math, but throw away any accessibility that would also make it possible for a disabled person to read this as well."

What is the use case for choosing the HTML only output mode? I don't really see why someone would need to just create the HTML without also having the MathML markup so that assisstive tech can make sense of it.

What I fear is that some users, who don't know about MathML and its relation to accessibility (which is most in my experience) will select the HTML only mode, since they dont know what MathML is and a natural reaction is to not use what you dont understand and then they will produce content that is unnecessarily inaccessible.

i get that HtML + MathML is the default, but I don't understand why there is an option for HTML only ode as well as I fear it might just make people creating inaccessible content possible.

As a blind person myself, I am regularly frustrated with inaccessible math on the web and only pretty recently have more websites started to have accessible math  mostly thanks to MathJax. And let's be honest, this has been almost entirely accidental on the part of the authors of the content, since most of them don't even know much about accessibility. Instead they just author content and if they accidently use a tool that produces MathML then the disabled community wins as an accidental side-effecct.

**Comment by SamKacer** [comment]
But let me also add that at least from what I've seen so far, you've done a absolutely stellar job at implementing great accessibility in the HTML + MathML output, so a ton of kudos is in ordr there!

**Comment by ylemkimon** [comment, marked as answer]
This is because authors may provide their own alternative text for the math. For example, Khan Academy uses [render-a11y-string](https://github.com/KaTeX/KaTeX/tree/master/contrib/render-a11y-string) plugin, which analyzes the math node to build the alternative text. I agree the options page lacks documentation regarding accessibility.

**Comment by SamKacer** [comment]
@ylemkimon thanks, that makes sense. I think that kind of info would be useful to have in the options doc, since otherwise it isn't very clear why someone would choose HTML instead of HTML + MathML and some users might just go for HTML only since MathML might be unfamiliar to them and the KISS principle.

Maybe part of the HTML only option there could also be something along the lines of
"Warning: choosing this option makes the output inaccessible. Choose this when implementing your own accessibility, such as when using RenderA11YString."
